src/library_pthread.js

```diff
@@ -211,11 +211,12 @@ var LibraryPThread = {
       // worker pool as an unused worker.
       worker.pthread_ptr = 0;
 
-#if ENVIRONMENT_MAY_BE_NODE
+#if ENVIRONMENT_MAY_BE_NODE && PROXY_TO_PTHREAD
       if (ENVIRONMENT_IS_NODE) {
-        // Once a pthread has finished and the worker becomes idle, mark it
-        // as weakly referenced so that its existence does not prevent Node.js
-        // from exiting.
+        // Once the proxied main thread has finished, mark it as weakly
```

Comment:
The comment here is specific to the proxied main  but the `_emscripten_thread_set_strongref` seems like a generic function.

Perhaps the comment could say 

```
// In case a thread was marked as strongly referenced using
// _emscripten_thread_set_strongref (for example, the proxied main thread)
// we need to ensure that its un-referenced when we the thread exits and
// the works is returned to the pool
```

Reply:
This `.unref()` is conditional guarded by `PROXY_TO_PTHREAD`, so I think the comment still makes sense. 

In other words, we can be sure that all threads are already weakly referenced when the worker is returned to the pool, expect for the "main" thread in `PROXY_TO_PTHREAD` mode; since that one is strongly referenced.